### PR TITLE
PYIC 3484 Refactor CiMitService to separate lib

### PR DESCRIPTION
**Pull request by rpayal**
## Proposed changes
Refactor CiMitService to separate lib

### What changed
Refactor code to separate cimitService as a separate dependent service.

### Issue tracking

- [PYIC-3484](https://govukverify.atlassian.net/browse/PYIC-3484)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-3484]: https://govukverify.atlassian.net/browse/PYIC-3484?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java [hunk 1]
```diff
@@ -9,22 +9,25 @@
...
 import com.nimbusds.jwt.SignedJWT;
 import com.nimbusds.oauth2.sdk.http.HTTPResponse;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Captor;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
+import org.mockito.Mockito;
```
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: Remove? We already have static imports for the relevent items from Mockito.

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java [hunk 2]
```diff
@@ -349,7 +352,7 @@ void getContraIndicatorsVCThrowsExceptionIfVCValidationFails() throws JsonProces
                         new InvokeResult()
                                 .withStatusCode(200)
                                 .withPayload(makeCiMitVCPayload(SIGNED_CONTRA_INDICATOR_VC)));
-        doThrow(
+        Mockito.doThrow(
```
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: We already have the static import.

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java [hunk 3]
```diff
@@ -9,22 +9,25 @@
 import com.nimbusds.jose.jwk.ECKey;
 import com.nimbusds.jwt.SignedJWT;
 import com.nimbusds.oauth2.sdk.http.HTTPResponse;
+import org.junit.jupiter.api.Assertions;
```
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: Remove? We already have the static imports we need from Assertions.

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java [hunk 4]
```diff
@@ -462,7 +465,7 @@ void getContraIndicatorsVCJwtWhenValidJWT()
                         GOVUK_SIGNIN_JOURNEY_ID, CLIENT_SOURCE_IP, TEST_USER_ID),
                 new String(request.getPayload().array(), StandardCharsets.UTF_8));
 
-        assertEquals(SIGNED_CONTRA_INDICATOR_VC, contraIndicatorsVCJwt.serialize());
+        Assertions.assertEquals(SIGNED_CONTRA_INDICATOR_VC, contraIndicatorsVCJwt.serialize());
```
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: We already have the static import

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorItemTest.java
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorItemTest.java:1]: Duplicated from lib but the class under test hasn't been

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorTest.java
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorTest.java:1]: Duplicated from lib but the class under test hasn't been

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorsTest.java
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorsTest.java:1]: Duplicated from lib but the class under test hasn't been

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/SharedClaimsResponseTest.java
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/SharedClaimsResponseTest.java:1]: Duplicated from lib but the class under test hasn't been

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/SharedClaimsTest.java
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/SharedClaimsTest.java:1]: Duplicated from lib but the class under test hasn't been

File: libs/cimit-service/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker
Comment (therealslimsladey) [inline libs/cimit-service/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Not convinced this is needed? Tests run happily without it.

**Review by therealslimsladey — CHANGES_REQUESTED** [review]
Sonar is also flagging low coverage on new code

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java (hunk 1 above)
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: done

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java (hunk 3 above)
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: done

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java (hunk 4 above)
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: done

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorTest.java
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorTest.java:1]: done

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorsTest.java
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorsTest.java:1]: done

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/SharedClaimsResponseTest.java
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/SharedClaimsResponseTest.java:1]: done

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/SharedClaimsTest.java
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/SharedClaimsTest.java:1]: done

File: libs/cimit-service/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker
Comment (rpayal) [reply libs/cimit-service/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: done

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorItemTest.java
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/domain/ContraIndicatorItemTest.java:1]: done

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java (hunk 2 above)
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: done

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java [hunk 5]
```diff
@@ -483,6 +484,49 @@ void getContraIndicatorsVCJwtWhenInvalidJWT() throws JsonProcessingException {
                                 TEST_USER_ID, GOVUK_SIGNIN_JOURNEY_ID, CLIENT_SOURCE_IP));
     }
 
+    @Test
+    void getContraIndicatorsVCJwtWhenInvalidCimitKey() throws JsonProcessingException {
```
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: Rename to `getContraIndicatorsVCJwtThrowsErrorWhenInvalidCimitKey`

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java [hunk 6]
```diff
@@ -483,6 +484,49 @@ void getContraIndicatorsVCJwtWhenInvalidJWT() throws JsonProcessingException {
...
+
+        assertThrows(
+                CiRetrievalException.class,
+                () ->
+                        ciMitService.getContraIndicatorsVCJwt(
+                                TEST_USER_ID, GOVUK_SIGNIN_JOURNEY_ID, CLIENT_SOURCE_IP));
+    }
+
+    @Test
+    void getContraIndicatorsVCJwtWhenNoVcBlock()
```
Comment (therealslimsladey) [inline libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: Rename to `getContraIndicatorsVCJwtThrowsErrorWhenNoVcBlock`

**Review by therealslimsladey — CHANGES_REQUESTED** [review]
couple of test renames

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java (hunk 5 above)
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: done

File: libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java (hunk 6 above)
Comment (rpayal) [reply libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/cimit/CiMitServiceTest.java]: done

**Review by therealslimsladey — APPROVED** [review]
lgtm

**Pull request merged**